src/value.rs

```diff
@@ -0,0 +1,763 @@
+                let big: num_bigint_04::BigInt = v.into();
+                call_python_constructor(py, "builtins", "int", (big.to_string(),), None)
+                    .map_err(DeserializationError::new)
+            }
+            NativeType::Double => {
+                let v = f64::deserialize(typ, val)?;
+                Ok(PyFloat::new(py, v).into_any())
+            }
+            NativeType::Float => {
+                let v = f32::deserialize(typ, val)?;
+                Ok(PyFloat::new(py, v as f64).into_any())
+            }
+            NativeType::Ascii | NativeType::Text => {
+                let v = <&str as DeserializeValue<'frame, 'metadata>>::deserialize(typ, val)?;
+                Ok(PyString::new(py, v).into_any())
+            }
+            NativeType::Boolean => {
+                let v = bool::deserialize(typ, val)?;
+                Ok(PyBool::new(py, v).to_owned().into_any())
+            }
+            NativeType::Date => {
+                let v = CqlDate::deserialize(typ, val)?;
+                let date: chrono_04::NaiveDate = v.try_into().map_err(DeserializationError::new)?;
+
+                py_call_method(
+                    py,
+                    "datetime",
+                    "date",
+                    "fromisoformat",
+                    (date.to_string(),),
+                    None,
+                )
+                .map_err(DeserializationError::new)
+            }
+            NativeType::Timestamp => {
+                let t = CqlTimestamp::deserialize(typ, val)?;
+                let millis = t.0;
+                let seconds = (millis as f64) / 1000.0;
+
+                let datetime_mod = py.import("datetime").map_err(DeserializationError::new)?;
+                let tz = datetime_mod
+                    .getattr("timezone")
+                    .map_err(DeserializationError::new)?
+                    .getattr("utc")
+                    .map_err(DeserializationError::new)?;
+
+                let kwargs = PyDict::new(py);
+                kwargs
+                    .set_item("tz", tz)
+                    .map_err(DeserializationError::new)?;
+
+                py_call_method(
+                    py,
+                    "datetime",
+                    "datetime",
+                    "fromtimestamp",
+                    (seconds,),
+                    Some(&kwargs),
+                )
+                .map_err(DeserializationError::new)
+            }
+            NativeType::Time => {
+                let t = CqlTime::deserialize(typ, val)?;
+
+                let time: time_03::Time = t.try_into().map_err(DeserializationError::new)?;
+
+                let hour = time.hour();
+                let minute = time.minute();
+                let second = time.second();
+                let micro = time.nanosecond() / 1000;
+
+                call_python_constructor(py, "datetime", "time", (hour, minute, second, micro), None)
+                    .map_err(DeserializationError::new)
```

Comment:
♻️ Could we maybe extract branches of this huge match to distinct functions? So that everything would get more modular.

Reply:
But it does make sense IMO to keep dispatcher (this function with the large `match`) separate from actual implementation. Jumpling between functions is only needed if you need to really read a particular implementation; what usually matters is ease of grasping the code top-down, which is easier if your functions are shorter and delegate things to intuitively extracted and named subroutines.

This is just my opinion, but I feel confident about it.

Reply:
Now the branches are quite concise, so it's no longer relevant.